replica/table.cc

```diff
@@ -1658,7 +1658,7 @@ future<> table::flush(std::optional<db::replay_position> pos) {
 }
 
 bool table::can_flush() const {
-    return _compaction_group->can_flush();
+    return std::ranges::any_of(compaction_groups(), std::mem_fn(&compaction_group::can_flush));
```

Comment:
What happens if table::can_flush() returns true for some compaction group, and then we call compaction_group::flush() even for compaction groups that can't?

Reply:
While extending the tests, I changed table::flush() to issue compaction::flush() only if compaction_group::can_flush() is true.  I'll patch table::flush() again with that change.